### ISSUE TITLE
Issue with drop visual indicators

### ISSUE THREAD

**Issue by HybridSolutions** [kommitters/editorjs-drag-drop#134]
Hi 

First of all, congrats for this addon! 

Using the latest version from CDN of both Editor and this plugin but the arrow is not centered. I have to put the CSS from index.css directly on my page to get it centered.  Here's what happens:

![image](https://github.com/kommitters/editorjs-drag-drop/assets/8672899/f4b907cb-8115-4b2f-aa6b-45f67c6b8abe)

Also, what is the correct way of changing the style of that dashed line without messing with the plugin code? Can it be done using CSS?

Thank you!

**Comment by HybridSolutions** [comment]
Meanwhile I managed to customize it with CSS to provide more accurate visual feedback. Here's a demo:

![demo](https://github.com/kommitters/editorjs-drag-drop/assets/8672899/090f0eb6-ba54-4b83-9696-b65760b3e2da)


The CSS I used:

```
.ce-block--drop-target .ce-block__content{
          border-top: 3px solid #ff6ab4 !important;
          border-bottom: 3px solid #ff6ab4 !important;
        }
        .ce-block--drop-target:last-child .ce-block__content{
          border-top: 1px dashed #a0a0a0 !important;
          border-bottom: 3px solid #ff6ab4 !important;
        }
        .ce-block--drop-target:first-child .ce-block__content{
          border-top: 3px solid #ff6ab4 !important;
          border-bottom: 1px dashed #a0a0a0 !important;
        }
        .ce-block--drop-target .ce-block__content:before {          
          top: 50%;
          left: -20px;
          height: 8px;
          width: 8px;
          border: solid #ff6ab4;
          border-width: 2px 2px 0 0;
        }
        .ce-block--drop-target:first-child .ce-block__content:before {
            top:0;
        }
        .ce-block--drop-target:last-child .ce-block__content:before {
            top:100%;
        }
        .ce-block--drop-target .ce-block__content:after {
          background: none;
        }
```

**Comment by antoineol** [comment]
Brilliant! For a better dragging experience, I added `pointer-events: none;` to the arrow (to avoid disabling the drag detection when hovering over the arrow) and replaced borders with outlines to avoid shifting the content.

I wish something as big as Editor.js had a nice native D&D experience, e.g. like [BlockNote](https://github.com/TypeCellOS/BlockNote) ([demo](https://blocknote-main.vercel.app/)) 😅

**Comment by HybridSolutions** [comment]
My issue is that if you use a 1px outline or change color, you still see that dashed line on the bottom. Should be possible to override it using CSS.

![image](https://github.com/kommitters/editorjs-drag-drop/assets/8672899/84a62ae0-89ef-4e1f-bbae-032a7f0bfeb2)

**Comment by antoineol** [comment]
For the default dashed line, that's what the second parameter is for:

```js
new DragDrop(editor, 'none');
```

(see my edit above)

Then we're free to style however we want.

I understand for the first and last, it's a good point. Ideally, we should have the same logic for every line, but there is no clue whether we're before or after the previous position :(

For consistency, I kept the arrow centered everywhere (to avoid the "is this a bug?" reaction), but that's a personal preference.

I agree for your preview! Or at least like Notion, which I find clearer than the default experience here. We could tune it a bit, but ideally it should be native to editorjs :)

**Comment by antoineol** [comment]
After playing a bit with it, here are the changes that give me an okay experience:

```js
// keep the default border style, no second argument, otherwise the last selector need to be updated accordingly.
new DragDrop(editor);
```

```css
.ce-block--drop-target .ce-block__content:before {
  /* Avoids a bug when dragging over the arrow */
  pointer-events: none;
}

.ce-block--drop-target .ce-block__content:after {
  /* Hides the default dashes */
  background: none;
}

.ce-block--drop-target
  .ce-block__content[style*='border-top: 1px dashed rgb(170, 170, 170);']:before {
  /* The fragile rule that moves the arrow, not only for the first element but all elements above the source */
  top: 0;
}
```

To go further,

- We could replace the border style to be more visible
- We could replace border with shadow (to simulate an outline in one side only), but it would require to change the source code, currently designed to use borders
- We could improve the selector to detect the direction, currently based on the strict style, which is [very fragile](https://stackoverflow.com/a/8426901/4053349). But it requires to change the plugin source code, again.

If changing the code, I would suggest to apply classes, e.g. on the dragged element (source) and the target element, to make style overrides easier.

**Comment by HybridSolutions** [comment]
I like it except the part we drag an element to the top. The position indicator appears below the first element and not after, but that probably just with code changes. 

![image](https://github.com/kommitters/editorjs-drag-drop/assets/8672899/8b3ab78b-42d9-4015-abdf-b0e41312ec22)
_Dragging the item 4 to the top._

**Comment by antoineol** [comment]
Weird, I have it at the correct position at the first and last position too. Same as you (for the first and last only). When trying my rules above, did you remove all others?

Anyway, if you have something working, that's good :)

**Comment by MarioRodriguezS** [comment]
@antoineol @HybridSolutions Thanks a lot for reporting and contributing to the issue. It was solved and the solution is available in the v1.1.14. The Readme was updated with the parameter to customize the indicator border, as described by @antoineol.

Cheers!
